### PR TITLE
feat: create user table and model

### PR DESCRIPTION
**Pull request by hanlinc27**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Update-Users-Table-Model-6627544f73424247ba1a01d934d7b510)

## Implementation description
* Create user table migration and update user model

## Steps to test
1. `docker exec -it community-fridge-kw_ts-backend_1 /bin/bash ` 
2. Run `node migrate up` 
image

## What should reviewers focus on?
* They can run the migration
* Is the code good 😎

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by linnall — DISMISSED** [review]

File: backend/typescript/migrations/create-users-table.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import { Migration } from "../umzug";
+
+export const up: Migration = async ({ context: sequelize }) => {
+  await sequelize.getQueryInterface().createTable("users", {
+    id: {
+      type: DataType.INTEGER,
+      allowNull: false,
+      primaryKey: true,
+    },
```
Comment (linnall) [inline backend/typescript/migrations/create-users-table.ts:11]: Does `unique: true` need to be added to this?

File: backend/typescript/models/user.model.ts [hunk 2]
```diff
@@ -3,15 +3,24 @@ import { Role } from "../types";
 
 @Table({ tableName: "users" })
 export default class User extends Model {
+  @Column({ type: DataType.INTEGER })
+  id!: number;
```
Comment (linnall) [inline backend/typescript/models/user.model.ts:7]: Does `@Unique` need to be added here?

File: backend/typescript/migrations/create-users-table.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    last_name: {
+      type: DataType.STRING,
+      allowNull: false,
+    },
+    phone_number: {
+      type: DataType.STRING,
+      allowNull: false,
+    },
+    firebase_uuid: {
```
Comment (Shehryar21) [inline backend/typescript/migrations/create-users-table.ts:24]: firebase_uuid would be unique I believe.

File: backend/typescript/models/user.model.ts [hunk 4]
```diff
@@ -3,15 +3,24 @@ import { Role } from "../types";
...
 
   @Column({ type: DataType.STRING })
-  auth_id!: string;
+  phone_number!: string;
+
+  @Column({ type: DataType.UUID })
+  firebase_uuid!: string;
 
-  @Column({ type: DataType.ENUM("User", "Admin") })
+  @Column({ type: DataType.ENUM("User", "Admin", "Donor") })
```
Comment (Shehryar21) [inline backend/typescript/models/user.model.ts:24]: Also add "Volunteer"

**Review by Shehryar21 — COMMENTED** [review]
PR looks good. Just take a look at the comments made. However just hold off on merging yet. I will look into how migrations and models are usually done in typescript backend just to make sure this is how it should be.

**Comment by Shehryar21** [comment]
Update: We actually do not need the migration files. Local db is updated when we do` docker-compose up` but for us that wasnt happening because node mailer was not configured and was causing the backend to crash. I will update the team in the next work session so please dont merge it.

**Pull request closed without merging**